DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/BulkUploadResultsViewModel.cs

```diff
@@ -0,0 +1,52 @@
+﻿namespace DigitalLearningSolutions.Web.ViewModels.TrackingSystem.Delegates
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DigitalLearningSolutions.Data.Services;
+
+    public class BulkUploadResultsViewModel
+    {
+        public BulkUploadResultsViewModel(BulkUploadResult bulkUploadResult)
+        {
+            Processed = bulkUploadResult.Processed;
+            Registered = bulkUploadResult.Registered;
+            Updated = bulkUploadResult.Updated;
+            Skipped = bulkUploadResult.Skipped;
+            Errors = bulkUploadResult.Errors.Select(x => (x.RowNumber, MapReasonToErrorMessage(x.Reason)));
+        }
+
+        public IEnumerable<(int RowNumber, string ErrorMessage)> Errors { get; set; }
+        public int ErrorCount => Errors.Count();
+        public int Processed { get; set; }
+        public int Registered { get; set; }
+        public int Updated { get; set; }
+        public int Skipped { get; set; }
+
+        private string MapReasonToErrorMessage(BulkUploadResult.ErrorReasons reason)
+        {
+            return reason switch
+            {
+                BulkUploadResult.ErrorReasons.InvalidJobGroupId =>
+                    "Job group ID was not valid. Please ensure a valid Job Group ID number is provided (use the 'Download Job Groups references' option for a list of valid IDs).",

```

Comment:
It's not covered by the ACs on the ticket but this message needs updating to reference the job groups sheet in the download instead. (Will probably need verifying with HEE what message they want)